uniconn.go

```diff
@@ -93,13 +98,26 @@ func (uc *UniConn) Write(b []byte) (n int, err error) {
 		return 0, ErrZeroLengh
 	}
 
+	var timeoutCtx context.Context
+	var timeoutCancel context.CancelFunc
+	if uc.writeTimeout > 0 {
+		timeoutCtx, timeoutCancel = context.WithTimeout(context.Background(), uc.writeTimeout)
```

Comment:
We can change background to uc.writeCtx so we can have one less case in select

Reply:
Yes, modified